### PR TITLE
[synthetic-monitoring-agent] Add deployment strategy

### PR DESCRIPTION
**Pull request by BlackDex**
Added a deploymentStrategy for the synthetic-monitoring-agent helm.

This allows more controle on how the pods will be deployed.
Used the same syntax as other helm charts already do.

### CODE REVIEW

File: charts/synthetic-monitoring-agent/templates/deployment.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ metadata:
   labels:
     {{- include "synthetic-monitoring-agent.labels" . | nindent 4 }}
 spec:
+  {{- with .Values.deploymentStrategy }}
+  strategy:
+    {{- toYaml . | trim | nindent 4 }}
```
Comment (Sheikh-Abubaker) [inline charts/synthetic-monitoring-agent/templates/deployment.yaml:10]:
```suggestion
    {{- toYaml . | nindent 4 }}
```

Comment (BlackDex) [reply charts/synthetic-monitoring-agent/templates/deployment.yaml:10]: Any reason to remove it? 
All other helm charts seen to have this if I'm correct.

Comment (Sheikh-Abubaker) [reply charts/synthetic-monitoring-agent/templates/deployment.yaml:10]: could you tell me any reason to include it ? any trailing whitespaces would already be trimmed.

Comment (BlackDex) [reply charts/synthetic-monitoring-agent/templates/deployment.yaml:10]: As mentioned, i copied it from other helm charts in this repo which all use the same.

Also, the `{{-` will not trim the input from value it self if I'm correct.

Comment (Sheikh-Abubaker) [reply charts/synthetic-monitoring-agent/templates/deployment.yaml:10]: Yeah but the CI lint test won't pass if there are trailing spaces in the `deployment strategy` value, so you don't have to worry about that.

Comment (BlackDex) [reply charts/synthetic-monitoring-agent/templates/deployment.yaml:10]: Ok, I'm fine with removing it, i just wanted it to match the rest of all the helm charts.

Comment (BlackDex) [reply charts/synthetic-monitoring-agent/templates/deployment.yaml]: Done

**Review by zanhsieh — APPROVED** [review]

**Review by Sheikh-Abubaker — APPROVED** [review]
LGTM!

**Pull request merged**